### DISCUSSION TITLE
Proposal: Path to `v3.0`

### DISCUSSION THREAD

**Discussion by nkramer44** [XRPLF/xrpl4j#187] [General]
We are currently working on improving the key management and signing functionality in xrpl4j, as well as contemplating a reorganization of our modules so that developers only have to import `xrpl4j-core`.

As discussed in #182, we would rather not rush to a version 3 and break existing functionality without providing a clear and well documented upgrade path from v2.x to v3.0. So, @sappenin and I propose the following sequence of work:

- [x] Get `v3-dev-preview` to a nearly complete state (See #170).
- [x] Create a `v2.next` branch off of main
- [ ] Iteratively commit `v3-dev-preview` branch enhancements into `v2.next` branch (keeping PRs small for future readability), keeping (and deprecating) old code so as not to break anything
- [x] Once we are happy with the `v2.next` branch, merge into main (no squash) and release a `v2.x`
- [x] Document `v2.x` new functionality
- [x] Create a `v2` maintenance branch for potential bug fixes in `v2` 
- [x] Begin considering `main` as the v3 branch (any releases from `main` will be `v3.x`)
- [x] Implement module changes suggested in #185 in the `main` branch
- [x] Document `v3.0.0` functionality (ie pom.xml changes for new module structure)
- [x] Delete all deprecated code
- [x] Release `v3.0.0`

**Comment by sappenin** [comment]
Closing this discussion now that v3 of xrpl4j is released.
